lib/hawkular/operations/operations_api.rb

```diff
@@ -322,6 +339,8 @@ def update_collection_intervals(hash, &callback)
     private
 
     def invoke_operation_helper(operation_payload, operation_name = nil, binary_content = nil, &callback)
+      synchronize { connect }
```

Comment:
I don't know where was the source of the exception. So, I'm guessing...

With your changes, I would expect `connect` to not raise an error even if the server rejects the connection or is down. And, then, I would expect the "connection refused" exception to be handled by `@ws.send` called in the later lines by calling the failure callback. I cannot figure any other way on how the callback may be invoked in case of any error. 

If my guessings are right, it's all ok. But as @mtho11 suggested in the other PR, it would be great if you can add a test to ensure that the code won't raise the "connection refused" exception, but invoke the failure callback. Specially, because @rubenvp8510 is doing work in replacing the websocket gem.